### PR TITLE
Add sub_nav macro and current page styling (fixes #10038)

### LINKED ISSUE #10038: Subnav link does apply styling to show when on active page
## Description

Part of Epic #10036

---

#### Success Criteria
- [ ] When sub nav is present on a page, the nav link should be active to reflect which page the user is on. 

## Screenshots

Active state example:

image

### PR DESCRIPTION
**Pull request by maureenlholland**
## Description

Adds sub_nav macro for standard HTML structure and dynamic `aria-current=page` attribute. This applies to our most consistent sub navigations.

Removes commented out Android Beta link in Release Notes sub nav: https://github.com/mozilla/bedrock/issues/10064

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/10038

## Testing

Check subnavs have expected text and links for the following

- [ ] Firefox Nightly & Beta: http://localhost:8000/en-US/firefox/channel/desktop/
- [ ] Firefox Desktop: http://localhost:8000/en-US/firefox/windows/
- [ ] Firefox Desktop Compare: http://localhost:8000/en-US/firefox/browsers/compare/
- [ ] Firefox Mobile: http://localhost:8000/en-US/firefox/browsers/mobile/
- [ ] Firefox Privacy: http://localhost:8000/en-US/firefox/privacy/safe-passwords/
- [ ] Diversity: http://localhost:8000/en-US/diversity/2021/
- [ ] Release Notes: http://localhost:8000/en-US/firefox/notes/
- [ ] VPN More: http://localhost:8000/en-US/products/vpn/more/what-is-a-vpn/
- [ ] VPN Desktop: http://localhost:8000/en-US/products/vpn/desktop/
- [ ] VPN Mobile: http://localhost:8000/en-US/products/vpn/mobile/
- [ ] mobile size, no js, the sub nav is visible in block display (no regression from https://github.com/mozilla/bedrock/pull/11451)

Test locally `py.test bedrock/base/tests/test_macros.py`

### CODE REVIEW

**Comment by maureenlholland** [comment]
to-do
- basic conditional render tests (title only, title with link, title with icon, title with icon and link, sub nav link has correct text, href, and cta name, sub nav list has correct order, current sub nav link has aria-current attribute)

**Comment by maureenlholland** [comment]
note: conflicts incoming with https://github.com/mozilla/bedrock/pull/11451/files#diff-9666439aa7d800b3fc044e9b47ade4a2bd02f52211f2c0973e8642a3bca43e37

**Comment by maureenlholland** [comment]
rebase and update in progress...

File: bedrock/firefox/templates/firefox/privacy/base.html [hunk 1]
```diff
@@ -20,31 +20,7 @@
...
-      {% if ftl_has_messages('sub-navigation-safe-passwords') %}
-        <li class="c-sub-navigation-item">
-          <a href="{{ url('firefox.privacy.passwords') }}" data-link-type="nav" data-link-position="subnav" data-link-name="Safe Passwords">{{ ftl('sub-navigation-safe-passwords') }}</a>
-        </li>
-      {% endif %}
-      </ul>
-    </div>
-  </div>
-</nav>
+  {% include 'firefox/privacy/includes/sub-nav.html' %}
```
Comment (reemhamz) [inline bedrock/firefox/templates/firefox/privacy/base.html:23]: On this file, in line 28, I see this block of code:
`{% block sub_nav %}{% endblock %}`
Should we remove it since you've already included the updated `side_nav `on this line?

**Review by reemhamz — APPROVED** [review]
Holy cow, I'm SO impressed with how you spearheaded this!
I don't believe we've managed macros like this before, and I learned a lot from reviewing your code. You implemented it very well, and very efficiently! GOOD JOB!!! 🚀✨

I left a couple comments to be answered, but if you think it's ready for a merge, go ahead! I'll leave it up to you!
r+wc!

File: bedrock/firefox/templates/firefox/privacy/base.html (hunk 1 above)
Comment (maureenlholland) [reply bedrock/firefox/templates/firefox/privacy/base.html:23]: Good catch! Yes, this appears to be dead code leftover from https://github.com/mozilla/bedrock/pull/9650/files#diff-5568b6af1f54d3948e2e55fb396463696562149cf7dc1719bdca2d903c9a2743

Will remove.

File: bedrock/base/templates/macros.html [hunk 2]
```diff
@@ -234,3 +234,55 @@ <h2 class="mzp-c-form-title">{{ form_title }}</h2>
     <img src="{{ src }}" alt="{{ alt }}" {% if class %}class="{{ class }}" {% endif %}{% if loading %}loading="{{ loading }}" {% endif %}{% if width %}width="{{ width }}" {% endif %}{% if height %}height="{{ height }}" {% endif %}/>
   {% endif %}
 {%- endmacro %}
+
+{% macro sub_nav(
+  title,
+  links
+) -%}
+
+{% if 'releasenotes' in request.path %}
```
[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/base/templates/macros.html:243]: Yeah, this is an odd one.

tl;dr: the if statement is here because there's a discrepancy between the request path and the link hrefs in release note pages and it felt simpler to manage here than in the note.html sub_nav call. 

request.path = /en-US/firefox/98.0.2/releasenotes/
link.href = /en-US/firefox/notes/

request.path = /en-US/firefox/99.0beta/releasenotes/
link.href = /en-US/firefox/beta/notes/

More detail:

I had no active page link styling on release note sub nav because link.href never === request.path. To get that active page link styling, I saw a few options:

1. Create a `current_page` variable in the macro and set that conditionally if we're on one of the release pages. Default to the `request.path` value.
2. Create a `current_page` param for the macro and set it conditionally in the `notes.html`template before making the sub_nav call. Param default is request.path.
3. Change the link hrefs in the notes sub_nav call to match the request path structure
4. Change the request path in the notes sub_nav call to match the link href structure

3 didn't scale well and had the potential to break links.
4 seemed like bad practice, with potential unintended consequences.

2 is still viable, I think, although I have to check if you can use context for a param default (the if statement might have to stay, i.e. if current_page is set, use that, else, use request.path). It does involve creating a new param, because otherwise there is no way to pass the matchable request path to the macro.

1 seemed simplest, as it didn't involve creating a new param that mostly wouldn't be used and kept the logic in one place that could handle more conditions if needed. BUT, if it causes confusion, maybe it's worth taking another look at option 2. Let me know what sounds best to you!

[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/base/templates/macros.html:242]: @reemhamz I'm going to go ahead and merge as-is for now, but we can definitely open a new, more focused issue to deal with the release page weirdness if it seems worthwhile
